### PR TITLE
RM-9287 Switch testmatrix to NET 8.0

### PR DESCRIPTION
**Pull request by Frazerus**
https://re-motion.atlassian.net/browse/RM-9287

### CODE REVIEW

File: Build/Build.cs [hunk 1]
```diff
@@ -275,19 +273,19 @@ public override void ConfigureTestMatrix (TestMatricesBuilder builder)
...
-            { Docker_Win_NET481, NET48, NoBrowser, SqlServer2019, Release, x64 },
-            { Docker_Win_NET48, NET48, NoBrowser, SqlServer2017, Release, x64 },
-            // { Docker_Win_NET48 , NET48, NoBrowser, SqlServer2016, Release, x64 },
+            { Docker_Win_NET8_0, NET8_0_WINDOWS, NoBrowser, SqlServer2022, Release, x64 },
+            { Docker_Win_NET8_0, NET8_0_WINDOWS, NoBrowser, SqlServer2019, Release, x64 },
+            { Docker_Win_NET8_0, NET8_0_WINDOWS, NoBrowser, SqlServer2017, Release, x64 },
+            // { Docker_Win_NET8_0 , NET8_0_WINDOWS, NoBrowser, SqlServer2016, Release, x64 },
             // { Docker_Win_NET50 , NET50, NoBrowser, SqlServer2022, Release, x64 },
             // { Docker_Win_NET50 , NET50, NoBrowser, SqlServer2019, Release, x64 },
             // { Docker_Win_NET50 , NET50, NoBrowser, SqlServer2017, Release, x64 },
```
Comment (a-ctor) [inline Build/Build.cs:291]: The docker image only support x64 and will fail for any x86 configuration. We should change this to use x64 instead.

**Review by a-ctor — COMMENTED** [review]
Changes look good. After changing `x86` to `x64` we need to check if the build runs through successfully. I think there are a few tests left that do not work with .NET. Also this PR is now on an old state of develop and needs to be rebased/cherry picked accordingly.

File: .nuke/parameters.ci-chrome.json [hunk 2]
```diff
@@ -3,9 +3,9 @@
   "Solution": "Remotion.sln",
   "Target": [ "CIBuild" ],
   "TestConfigurations": [ "Debug", "Release" ],
-  "TestExecutionRuntimes": [ "EnforcedLocalMachine" ],
+  "TestExecutionRuntimes": [ "EnforcedLocalMachine(Docker_Win_NET8_0)" ],
```
Comment (MichaelKetting) [inline .nuke/parameters.ci-chrome.json:6]: @a-ctor when using the original value ("EnforcedLocalMachine", without the Docker-Argument), I get a warning in Rider that this isn't a supported value. When I update to include the Docker-Argument, I get a runtime error 
```
Target "CreateTestMatrix" has thrown an exception
 Target "CreateTestMatrix" has thrown an exception
    [ERR] Target CreateTestMatrix has thrown an exception
    System.InvalidOperationException: The supplied test dimension 'EnforcedLocalMachine(Docker_Win_NET8_0)' is not supported.
       at Remotion.BuildScript.Test.SupportedTestDimensions.ParseTestDimensionValuesOrDefault[T](String[] values)
       at Remotion.BuildScript.RemotionBuild.ConfigureEnabledTestDimensions(EnabledTestDimensionsBuilder enabledTestDimensions)
       at Build.ConfigureEnabledTestDimensions(EnabledTestDimensionsBuilder enabledTestDimensions) in C:\Build\28109e4c1456d59b\working\build\Build.cs:line 231
       at Remotion.BuildScript.Components.ITestMatrix.<get_CreateTestMatrix>b__10_1()
       at Nuke.Common.Execution.BuildExecutor.<>c.<Execute>b__4_2(Action x) in /_/source/Nuke.Build/Execution/BuildExecutor.cs:line 119
       at System.Collections.Generic.List`1.ForEach(Action`1 action)
       at Nuke.Common.Execution.BuildExecutor.Execute(NukeBuild build, ExecutableTarget target, IReadOnlyCollection`1 previouslyExecutedTargets, Boolean failureMode) in /_/source/Nuke.Build/Execution/BuildExecutor.cs:line 119
```
     
       
Any thoughts?

**Review by MichaelKetting — COMMENTED** [review]
partial review

File: .nuke/parameters.ci-chrome.json (hunk 2 above)
Comment (a-ctor) [reply .nuke/parameters.ci-chrome.json:6]: It should stay as `EnforcedLocalMachine`, the "not supported value" is a bug in the schema generation.

File: .nuke/build.schema.json [hunk 3]
```diff
@@ -161,10 +161,8 @@
           "items": {
             "type": "string",
             "enum": [
-              "Docker_Win_NET48",
-              "Docker_Win_NET481",
               "Docker_Win_NET8_0",
-              "EnforcedLocalMachine(Docker_Win_NET6_0)",
+              "EnforcedLocalMachine(Docker_Win_NET8_0)",
```
Comment (MichaelKetting) [inline .nuke/build.schema.json:165]: Here and in Build.cs we still had net-6 mentioned. I fixed that one.
@a-ctor 
cc @Frazerus

File: Build/Customizations/DatabaseTestSetup.cs [hunk 4]
```diff
@@ -101,15 +101,9 @@ public void ExecuteTests (TestExecutionContext context, Action<TestExecutionCont
     appConfig.SetAppSetting("Password", password);
 
     appConfig.WriteToFile(configFile);
+    File.Copy(configFile, configFile.Parent / "testhost.dll.config", true);
+    File.Copy(configFile, configFile.Parent / "testhost.x86.dll.config", true);
```
Comment (MichaelKetting) [inline Build/Customizations/DatabaseTestSetup.cs:105]: @a-ctor copying the config for `dotnet test` here and in WebTestingTestSetup.cs.

**Comment by MichaelKetting** [comment]
@a-ctor two broken tests (probably a recent CoreForms upgrade:
```
Assert.That(caughtException, expression)
  Expected: <System.Web.HttpUnhandledException>
  But was:  <System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Web.PerfCounters.IncrementCounter(AppPerfCounter counter) in /_/src/CoreForms.Web/PerfCounters.cs:line 205
   at System.Web.UI.Page.ProcessRequestMain(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint) in /_/src/CoreForms.Web/UI/Page.cs:line 4854
   at System.Web.UI.Page.ProcessRequest(Boolean includeStagesBeforeAsyncPoint, Boolean includeStagesAfterAsyncPoint) in /_/src/CoreForms.Web/UI/Page.cs:line 4445
   at System.Web.UI.Page.ProcessRequest() in /_/src/CoreForms.Web/UI/Page.cs:line 4422
   at System.Web.UI.Page.ProcessRequestWithNoAssert(HttpContext context) in /_/src/CoreForms.Web/UI/Page.cs:line 4392
   at System.Web.UI.Page.ProcessRequest(HttpContext context) in /_/src/CoreForms.Web/UI/Page.cs:line 4382
   at Remotion.Web.UnitTests.Core.ExecutionEngine.WxePageTests.InfrastructureExceptionHandling_WxePageTest.<>c__DisplayClass2_0.<ThrowAnyException_WrappedsInHttpUnhandledException>b__0() in D:\Development\Remotion\Framework\Remotion\Web\UnitTests\Core\ExecutionEngine\WxePageTests\InfrastructureExceptionHandling_WxePageTest.cs:line 61
   at NUnit.Framework.Assert.Throws(IResolveConstraint expression, TestDelegate code, String message, Object[] args)>

   at Remotion.Web.UnitTests.Core.ExecutionEngine.WxePageTests.InfrastructureExceptionHandling_WxePageTest.ThrowAnyException_WrappedsInHttpUnhandledException() in D:\Development\Remotion\Framework\Remotion\Web\UnitTests\Core\ExecutionEngine\WxePageTests\InfrastructureExceptionHandling_WxePageTest.cs:line 61
```

**Review by a-ctor — APPROVED** [review]

File: SharedSource/UnitTests/Development/Core/UnitTesting/PEVerifierTest.cs [hunk 5]
```diff
@@ -11,6 +11,7 @@
 // ReSharper disable once CheckNamespace
 namespace Remotion.Development.UnitTests.Core.UnitTesting
 {
+  /*
```
[the remark this answers is not in the document]
Comment (MichaelKetting) [reply SharedSource/UnitTests/Development/Core/UnitTesting/PEVerifierTest.cs:14]: They get dropped when dropping .net Framework. Or, more precisely, with RM-9300 now. might that I merge that branch first.

**Pull request merged**